### ISSUE TITLE
Building Opus fails with `configure: error: could not make ./config.status`

### ISSUE THREAD

**Issue by abhayghatpande** [eofster/Telephone#326]
Facing a strange build error on 10.12. None of the packages Opus, PJSIP, and LibreSSL will build. All the configure steps exit with the same error. 

```
configure: creating ./config.status
configure: error: could not make ./config.status
```

Xcode 8.1 is installed with command line tools.

```
$ gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin16.1.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

I realize that this is not something related to Telephone directly, but any pointers will help.  Thanks!

**Comment by eofster** [comment]
Maybe the paths you specify in the command line have unescaped spaces or quotations?

**Comment by abhayghatpande** [comment]
Nope, I specified the full path `/Users/<username>/Workspace/Telephone/ThirdParty/Opus`.

The entire configure script runs correctly, up to the point of creating the config.status and the Makefile, where it fails. I installed the command line tools separately and switched the path using `xcode-select -s /Library/Developers/CommandLineTools`. Also updated the PATH to reflect gcc from there. But the result is the same.

So a little bit of progress. The culprit seems to be `sed`. I noticed that I had the gnu version of sed in the PATH. When I removed that to use the default `sed`, I got an error: 
```
configure: creating ./config.status
sed: 1: "s/.*%!_!# !%!_!#  _%!_! ...": unterminated substitute pattern
configure: error: could not make ./config.status
```

This is true for the other packages  too. There is a large `sed` replacement pattern and that's choking up for some reason in the `aconfigure` and `configure` scripts. The question is why and how to get around it. 

The curious thing is that using gnu-sed from Homebrew does not help, and it exits without even printing the error above generated by OS X `sed`.

**Comment by eofster** [comment]
My `xcode-select -p` command shows this path to the developer directly.

```
/Applications/Xcode.app/Contents/Developer
```

Does it work if you set the same path?

**Comment by abhayghatpande** [comment]
```
$ sudo xcode-select -s /Applications/Xcode.app/Contents/Developer
$ xcode-select -p
/Applications/Xcode.app/Contents/Developer
$ gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin16.1.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Then,
```
./configure --prefix=/Users/abhay/Workspace/Telephone/ThirdParty/Opus --disable-shared CFLAGS='-O2 -mmacosx-version-min=10.10'
:
:
:
configure: creating ./config.status
sed: 1: "s/.*%!_!# !%!_!#  _%!_! ...": unterminated substitute pattern
configure: error: could not make ./config.status
```

Same result! Your `sed` is the system default?

**Comment by eofster** [comment]
Yes, I have only default tools. Clean latest macOS with the latest Xcode.

**Comment by abhayghatpande** [comment]
I cannot figure out what might be tripping up `sed`.  Terminal encoding? Locale settings?

**Comment by eofster** [comment]
Which path does the command `xcrun -f sed` print out? Mine is `/usr/bin/sed`.

**Comment by abhayghatpande** [comment]
The same for me. So weird ... Maybe there is some test script for `sed` to make sure that your and my `sed` are compatible?? I've read and heard of different `sed`s out there that accept different arguments/options.

**Comment by eofster** [comment]
I have no idea, sorry.

**Comment by abhayghatpande** [comment]
Hi Alexey,

Could you do me a favour and do a `man sed` and see the year at the bottom? Mine says 2005. I'm afraid it's an old version.

Would it be possible to somehow send me your `sed` executable?

**Comment by abhayghatpande** [comment]
Something is seriously wrong in my setup. Running `./configure --help` displays nothing. The command just exits. On Linux, the same command displays the usual help about `configure`. This is super strange. Maybe I need to reinstall my OS ...

**Comment by eofster** [comment]
My `man sed` also has year 2005. And here’s the output of `shasum /usr/bin/sed`:
```
bc498bf7fbfa76a736e6ca039e09cea38d80c616  /usr/bin/sed
```

**Comment by abhayghatpande** [comment]
oh no! mine is the same. which means it's not a `sed` issue?? what could it be? i've run out of ideas to debug this.

**Comment by Sija** [comment]
@abhayghatpande Maybe it's an issue with shell you're using and its special chars escaping strategy.

**Comment by abhayghatpande** [comment]
My shell is just bash, nothing fancy. The error statement `could not make ./config.status` occurs at 3 places in the `configure` file.
```
{
  echo "cat >conf$$subs.awk <<_ACEOF" &&
  echo "$ac_subst_vars" | sed 's/.*/&!$&$ac_delim/' &&
  echo "_ACEOF"
} >conf$$subs.sh ||
  as_fn_error $? "could not make $CONFIG_STATUS" "$LINENO" 5
ac_delim_num=`echo "$ac_subst_vars" | grep -c '^'`
ac_delim='%!_!# '
for ac_last_try in false false false false false :; do
  . ./conf$$subs.sh ||
    as_fn_error $? "could not make $CONFIG_STATUS" "$LINENO" 5

  ac_delim_n=`sed -n "s/.*$ac_delim\$/X/p" conf$$subs.awk | grep -c X`
  if test $ac_delim_n = $ac_delim_num; then
    break
  elif $ac_last_try; then
    as_fn_error $? "could not make $CONFIG_STATUS" "$LINENO" 5
  else
    ac_delim="$ac_delim!$ac_delim _$ac_delim!! "
  fi
done
``` 
I'm wondering how can I find which one is getting triggered?

**Comment by abhayghatpande** [comment]
As I'm not able to build it, is it possible to download Telephone executable/app somewhere?

**Comment by eofster** [comment]
I distribute it only via Mac App Store.

**Comment by abhayghatpande** [comment]
Got it! And you shall have my patronage, Alexey! Thank you for creating this. If I ever figure out the issue with the build, I will post it here.

**Comment by eofster** [comment]
Thank you Abhay! I’m closing this issue for now because it seems to be specific to only one computer. But feel free to comment here again if you have more details. I will reopen it, if needed.

**Comment by abhayghatpande** [comment]
So it works, but I can't say it's solved because I don't have the exact cause of the problem.
It turns out that this could be a PATH related issue. I came across this [post]( http://stackoverflow.com/questions/34430744/error-while-running-rbenv-install-on-os-x-10-11-2-config-status-error-could-n) that mentions the possibility of a PATH causing problems.
Though I could see nothing related to ruby or java or python on my PATH, I had previously played around with `rbenv`, `jenv`, and `pyenv`. The only thing I current had was `npm-packages` on my PATH.
So I just commented out everything related to PATH on my `~/.bash_profile` and tried the `configure` again, and voila!, it works! The `config.status` file is created, `configure --help` also displays the help info, etc.
I've yet to figure out what was the offending segment on my PATH. But for the moment, I'm happy that I'm able to start building at the command line again! Hope this helps someone facing similar problems in the future.
